biogeophys/LeafBiophysicsMod.F90

```diff
@@ -0,0 +1,2269 @@
+    ! for the bisection algorithm. It finds the values associated with minimum
+    ! and maximum conductance when equating the source and sink limitations
+    ! on net photosynthesis.
+
+    ! input
+    integer, intent(in)  :: ft             ! plant functional type index
+    real(r8), intent(in) :: vcmax          ! maximum rate of carboxylation (umol co2/m**2/s)
+    real(r8), intent(in) :: jmax           ! maximum electron transport rate (umol electrons/m**2/s)
+    real(r8), intent(in) :: kp             ! initial slope of CO2 response curve (C4 plants)
+    real(r8), intent(in) :: co2_cpoint     ! CO2 compensation point (Pa)
+    real(r8), intent(in) :: mm_kco2        ! Michaelis-Menten constant for CO2 (Pa)
+    real(r8), intent(in) :: mm_ko2         ! Michaelis-Menten constant for O2 (Pa)
+    real(r8), intent(in) :: can_co2_ppress ! Partial pressure of CO2 near the leaf surface (Pa)
+    real(r8), intent(in) :: can_o2_ppress  ! Partial pressure of O2 near the leaf surface (Pa)
+    real(r8), intent(in) :: can_press      ! Air pressure near the surface of the leaf (Pa)
+    real(r8), intent(in) :: lmr            ! leaf maintenance respiration rate (umol CO2/m**2/s)
+    real(r8), intent(in) :: par_abs        ! par absorbed per unit leaf area [umol photons/m2leaf/s ]
+    real(r8), intent(in) :: gb             ! leaf boundary layer conductance (umol H2O/m**2/s)
+    real(r8), intent(in) :: gs0            ! stomatal intercept
+    
+    ! output
+    real(r8), intent(out) :: ci_max  ! Intracellular Co2 at maximum conductance [Pa]
+    real(r8), intent(out) :: ci_min  ! Intracellular CO2 at minimum conductance [Pa]
+    
+    ! Intermediate terms
+    real(r8), dimension(3) :: ci  ! Ci for Rubisco,RuBP and PEP respectively
+    real(r8), dimension(3) :: ag  ! Agross for Rubisco, Rubp and PEP
+
+    ! These are compound terms used to solve the equation that balances
+    ! net assimilation with the gradient flux equation
+    real(r8) :: a,b,c,d,e,f,g
+    real(r8) :: ap,ac,aj,ai
+    real(r8) :: Je                ! Electron tranport rate (umol e/m2/s)
+    real(r8) :: rmin ,rmax        ! Maximum and minimum resistance [s/umol h2o/m2]
+    real(r8) :: aquad,bquad,cquad,r1,r2
+    logical :: err
+    
+    ! Minimum possible resistance (with a little buffer)
+    rmin = 0.75_r8*h2o_co2_bl_diffuse_ratio/gb
+    ! Maximum possible resistance (with a littel buffer)
+    rmax = 1.25_r8*(h2o_co2_bl_diffuse_ratio/gb + h2o_co2_stoma_diffuse_ratio/gs0)
+    
+    if (lb_params%c3psn(ft) == c4_path_index)then
+
+       ! Maximum conductance (minimum resistance)
+       
+       ag(1) = vcmax
+       ag(2) = AgrossRuBPC4(par_abs)
+
+       ! C4: Rubisco-limited photosynthesis
+       ac = vcmax
+       
+       ! C4: RuBP-limited photosynthesis
+       aj = AgrossRuBPC4(par_abs)
+       
+       !! C4: PEP carboxylase-limited (CO2-limited)
+       !ap = AgrossPEPC4(ci,kp,can_press)
```

Comment:
Maybe I missed/forgot some discussions here, but is there a reason to define ap in current line 622 instead of using the `AgrossPEPC4` function?

Reply:
nope, good catch, the math was the same, but its better to use the function, will fix